### PR TITLE
[DistillationTrainer refactor] Emit prompt_ids / prompt_mask / completion_ids

### PR DESCRIPTION
**Pull request by qgallouedec**
*Item 20 (Group B) — stacked on PR 19. Part of #6449.*

Third step of the layout migration (add → switch → delete). Emit GRPO's keys alongside the existing ones; nothing consumes them yet (item 21 switches consumers, item 22 deletes the old helpers).

- Both generation callers and the collator now emit `prompt_ids` (= `prompts`), `prompt_mask` (= `prompt_attention_mask`), and `completion_ids` (the completion region of `input_ids`; empty for the prompt-only collator).
- Buffer-side emission is throwaway (replaced by GRPO generation at item 29).

New test asserts `cat(prompt_ids, completion_ids) == input_ids` and that the prompt keys mirror the existing tensors.

Verified: 53 tests pass (distillation + server); ruff clean.

---

> [!NOTE]
> **Low Risk**
> Additive tensor keys only; loss and training logic still use the legacy fields, so behavior is unchanged aside from slightly larger batch dicts.
> 
> **Overview**
> **DistillationTrainer** batch dicts gain GRPO-style **`prompt_ids`**, **`prompt_mask`**, and **`completion_ids`** in addition to the existing `prompts` / `prompt_attention_mask` layout. Nothing reads the new keys yet; this is the “add alongside” step before consumers switch and old helpers are removed.
> 
> The **`_DistillationCollator`** return value adds the three keys (`prompt_ids`/`prompt_mask` mirror the prompt tensors; `completion_ids` is the tail slice of `input_ids`, empty for prompt-only collate). After on-policy generation, **`_generate_with_model`** and **`_store_completions_in_buffer`** populate the same fields on buffered microbatches.
> 
> A new integration test captures inputs in **`compute_loss`** and asserts the keys exist, that **`cat(prompt_ids, completion_ids) == input_ids`**, and that the prompt keys match **`prompts`** / **`prompt_attention_mask`**.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 69bbafcbe86844085931721cf2830efd64285163. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: trl/experimental/distillation/distillation_trainer.py [hunk 1]
```diff
@@ -773,6 +776,9 @@ def _generate_with_model(self, slices: list[dict[str, torch.Tensor | Any]], on_p
                 updated["attention_mask"] = new_attention_mask
                 updated["labels"] = new_labels
                 updated["completion_mask"] = new_completion_mask
+                updated["prompt_ids"] = slice_inputs["prompts"]
+                updated["prompt_mask"] = prompt_mask
+                updated["completion_ids"] = generated_tokens[:, prompt_width:]
```
[the remark this answers is not in the document]
Comment (qgallouedec) [reply trl/experimental/distillation/distillation_trainer.py:781]: again, this is intended

**Review by cmpatino — APPROVED** [review]
LGTM!
